src/OpenLoco/src/Ui/Windows/Vehicle.cpp

```diff
@@ -2878,22 +2880,256 @@ namespace OpenLoco::Ui::Windows::Vehicle
             return args;
         }
 
+        // 0x004B5BB9
+        static ViewportInteraction::InteractionArg stationLabelAdjustedInteraction(const Vehicles::VehicleHead& head, int16_t var_842, StationId stationId, ViewportInteraction::InteractionArg interaction)
+        {
+            auto* station = StationManager::get(stationId);
+            if (station == nullptr)
+            {
+                return ViewportInteraction::kNoInteractionArg;
+            }
+            if (station->owner != CompanyManager::getControllingId())
+            {
+                return ViewportInteraction::kNoInteractionArg;
+            }
+
+            _hoveredStationId = stationId;
+            World::setMapSelectionFlags(World::MapSelectionFlags::hoveringOverStation);
+            ViewportManager::invalidate(station);
+
+            {
+                auto args = FormatArguments::mapToolTip(StringIds::click_to_insert_new_order_stop_at);
+                args.push(station->name);
+                args.push(station->town);
+            }
+
+            if (var_842 != 0)
+            {
+                uint32_t targetOffset = 0U;
+                if (var_842 < 0)
+                {
+                    Vehicles::OrderRingView orders(head.orderTableOffset);
```

Comment:
you can move this line, the next line and line 2917 outside of the if/else as these statements run in both if and else.

Reply:
```
Vehicles::OrderRingView orders(head.orderTableOffset);
auto lastOrder = orders.begin();
if (var_842 < 0)
{
    while ((lastOrder + 1) != orders.end())
    {
        lastOrder++;
    }
    targetOffset = lastOrder->getOffset();
}
else
{
    while ((lastOrder + 1) != orders.end() && var_842 != 0)
    {
        lastOrder++;
        var_842--;
    }
}
targetOffset = lastOrder->getOffset();
```